env0/resource_team_environment_assignment.go

```diff
@@ -39,15 +39,15 @@ func resourceTeamEnvironmentAssignment() *schema.Resource {
 	}
 }
 
-func resourceTeamEnvironmentAssignmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	var newAssignment client.AssignTeamRoleToEnvironmentPayload
-	if err := readResourceData(&newAssignment, d); err != nil {
+func resourceTeamEnvironmentAssignmentCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
```

Comment:
A general comment, I was triggered by this line.
**User** Environment Assignment has the same structure and API behind the scenes. 
using `userId` instead of `teamId` and maybe the `path` is different (in 3 of the APIs).

We probably should commonize / reuse / sync the structure of your changes to that scope too.
is there anything to sync there?

Reply:
Yes - we could make an extra step.
Probably need to repeat this whole exercise for users.

right now user assignments are using different APIs (project, environment, organization). Therefore, if they're merged to one API (?), we can do that extra step afterwards.

Reply:
> repeat this whole exercise for users.

currently, only the **env** level has the new api.
so not urgent, anyways Org and Project will have the old apis for now.